main/ui/src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyFactory.java

```diff
@@ -72,8 +88,14 @@ String createRecoveryKey(byte[] rawKey) {
 	 */
 	public void resetPasswordWithRecoveryKey(Path vaultPath, String recoveryKey, CharSequence newPassword) throws IOException, IllegalArgumentException {
 		final byte[] rawKey = decodeRecoveryKey(recoveryKey);
-		try {
-			CryptoFileSystemProvider.restoreRawKey(vaultPath, MASTERKEY_FILENAME, rawKey, PEPPER, newPassword);
+		try (var masterkey = new Masterkey(rawKey)) {
+			Path masterkeyPath = vaultPath.resolve(MASTERKEY_FILENAME);
+			if (Files.exists(masterkeyPath)) {
+				byte[] oldMasterkeyBytes = Files.readAllBytes(masterkeyPath);
```

Comment:
Also here, like in the ChangePasswordCOntroller the user should be informed about the silent backup.

Additionally, what if the masterkey is loaded from somewhere else? We only set the recovery key inside the vault, not the ones stored in other places.

Reply:
This is indeed something that needs to be adjusted, once we add alternate key sources. So far, the keyloader api is "readonly", so restoring keys is not currently supported via a common interface. Needs to be addressed in a separate change.